src/states/ingame/IngameClient.js

```diff
@@ -0,0 +1,35 @@
+import State from 'states/state'
+import eventBus from 'states/events'
+import Graphics from './graphics'
+import * as events from './events'
+
+class IngameClient extends State {
+
+  create() {
+    this.graphics = new Graphics()
+    this.graphics.create()
+  }
+
+  connect(socket) {
+    eventBus.on(events.USER_END_GAME, this.endGame(socket))
+    socket.on(events.SERVER_END_GAME, this.onEndGame)
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  disconnect(socket) {
+    socket.removeAllListeners()
```

Comment:
Maybe it could be placed in `State.disconnect` ? It seems that you call `socket.removeAllListeners` in every classes that inherits from `State`